UnitsNet.Serialization.JsonNet/UnitsNetJsonConverter.cs

```diff
@@ -231,5 +240,52 @@ protected virtual bool CanConvertNullable(Type objectType)
         }
 
         #endregion
+
+        #region "MultiDimensional Array Helpers"
+
+        private static Array ConvertArrayElements<TResult>(Array array)
```

Comment:
I am not too familiar with multi dimensional arrays, so I also found the method names `FirstIndex` and `LastIndex` puzzling, expecting an integer value returned.

Could it help to rename to `FirstIndices` and `LastIndices` maybe? Just to clarify that we are getting one or more indicies depending on rank.

Reply:
> I think it would help readability to move these helper methods to a helper class, something like `MultiDimensionalArrayHelpers`, make the class and methods `internal` and add xmldoc to them describing a shortly what they do.

Done. Created it under the 'Internal' folder in the project.

> We could also add unit tests on the helper methods, but I assume they are already well covered by serialization tests.

I think these methods are well covered by the serialization tests themselves, if anything was amiss with them, I *believe* it would be impossible for the serialization tests to pass.  I can't really think of any non-trivial tests to create for them.  If you'd like there to be explicit tests for them for the sake of completeness though, I can add some in.  If so, where would you like the tests to reside?


> I am not too familiar with multi dimensional arrays, so I also found the method names `FirstIndex` and `LastIndex` puzzling, expecting an integer value returned.
> 
> Could it help to rename to `FirstIndices` and `LastIndices` maybe? Just to clarify that we are getting one or more indicies depending on rank.


I think `FirstIndex` and `LastIndex` are the appropriate terms.  The singular 'index' meaning the pointer to a specific element in a multidimensional array, whereas conversely, I would personally expect 'Indices' to refer to the locations of multiple elements in an array.

Looking at the Microsoft documentation, e.g., [SetValue](https://docs.microsoft.com/en-us/dotnet/api/system.array.setvalue?view=netcore-3.1), specifically, one example states `// Sets the element at index 1,3.`, referring to it in the singular.  Hopefully the comments on the functions cleared up their purpose.